packages/broker/src/plugins/operator/MaintainOperatorValueHelper.ts

```diff
@@ -1,33 +1,127 @@
+    /**
+     * Find the sum of unwithdrawn earnings in Sponsorships (that the Operator must withdraw before the sum reaches a limit),
+     * SUBJECT TO the constraints, set in the OperatorServiceConfig:
+     *  - only take at most maxSponsorshipsCount addresses (those with most earnings), or all if undefined
+     *  - only take sponsorships that have more than minSponsorshipEarnings, or all if undefined
+     * @param operatorContractAddress
+     */
+    async getUnwithdrawnEarningsOf(operatorContractAddress: EthereumAddress): Promise<UnwithdrawnEarningsData> {
+        const operator = new Contract(operatorContractAddress, operatorABI, this.config.signer) as unknown as Operator
+        const minSponsorshipEarningsWei = BigNumber.from(this.config.minSponsorshipEarnings ?? 0)
+        const { sponsorshipAddresses: allSponsorshipAddresses, earnings } = await operator.getEarningsFromSponsorships()
+
+        const sponsorships = allSponsorshipAddresses
+            .map((address, i) => ({ address, earnings: earnings[i] }))
+            .filter((sponsorship) => sponsorship.earnings.gte(minSponsorshipEarningsWei))
+            .sort((a, b) => Number(b.earnings.sub(a.earnings).toBigInt())) // TODO: after Node 20, use .toSorted() instead
+            .slice(0, this.config.maxSponsorshipsCount) // take all if maxSponsorshipsCount is undefined
+        const sponsorshipAddresses = sponsorships.map((sponsorship) => sponsorship.address as EthereumAddress)
+
+        const approxPoolValue = (await operator.totalValueInSponsorshipsWei()).toBigInt()
+        const sumDataWei = sponsorships.reduce((sum, sponsorship) => sum.add(sponsorship.earnings), BigNumber.from(0)).toBigInt()
+        const fraction = sumDataWei * ONE_ETHER / approxPoolValue
+
+        return { sumDataWei, fraction, sponsorshipAddresses }
+    }
+
+    async getMyUnwithdrawnEarnings(): Promise<UnwithdrawnEarningsData> {
+        return this.getUnwithdrawnEarningsOf(this.config.operatorContractAddress)
+    }
+
+    async withdrawMyEarningsFromSponsorships(sponsorshipAddresses: EthereumAddress[]): Promise<void> {
+        await (await this.operator.withdrawEarningsFromSponsorships(sponsorshipAddresses)).wait()
+    }
+
+    async triggerWithdraw(targetOperatorAddress: EthereumAddress, sponsorshipAddresses: EthereumAddress[]): Promise<void> {
+        await (await this.operator.triggerAnotherOperatorWithdraw(targetOperatorAddress, sponsorshipAddresses)).wait()
     }
 
-    async updateApproximatePoolValueOfSponsorships(sponsorshipAddresses: string[]): Promise<void> {
-        await (await this.operator.updateApproximatePoolvalueOfSponsorships(sponsorshipAddresses)).wait()
+    private async getOperatorAddresses(requiredBlockNumber: number, queryFilter: string): Promise<EthereumAddress[]> {
+        const createQuery = () => {
+            return {
+                query: `
+                    {
+                        operators${queryFilter} {
```

Comment:
Parameter `queryFilter` left blank in actual uses of method. I would remove parameter.
```suggestion
    private async getOperatorAddresses(requiredBlockNumber: number): Promise<EthereumAddress[]> {
        const createQuery = () => {
            return {
                query: `
                    {
                        operators {
```

Reply:
there should be pagination coming up (for the random selection), so the queryFilter is probably needed

Reply:
The pagination is automatically handled if we inject the the parameters of `createQuery` (`lastId: string, pageSize: number`) to line `101`. (see e.g. https://github.com/streamr-dev/network/blob/f782bb9155e2d399fe5424d56e9a6b411945bc27/packages/broker/src/plugins/operator/MaintainTopologyHelper.ts#L107)

Reply:
removed param